### PR TITLE
update ember-cli-version-checker

### PR DESCRIPTION
**Pull request by arjunkathuria**
PR for issue #71 

cc:- @rwjblue

### CODE REVIEW

**Comment by rwjblue** [comment]
:+1:

**Comment by arjunkathuria** [comment]
@runspired tests fail here, but not related to this change, is this the correct branch to merge in?

**Comment by runspired** [comment]
I believe so, but it's been a while, and to simplify my OSS contributions I think I want to shift this project back to "everything into master" like other Ember repos do. I would start with it here but merge into master and set master to be the primary branch from now on.

**Comment by arjunkathuria** [comment]
@eriktrom this PR is similar to #74, which already got merged.

lets bump up `"ember-cli-version-checker": "^1.2.0"` or newer while we're at it.

**Comment by eriktrom** [comment]
@arjunkathuria - thanks for the ping - didn't realize we had a huge drift between develop and npm - before randomly start merging into master, i opened #77 to sync things up, then i'll base this off master and merge it in (along with at least #76)

**Comment by eriktrom** [comment]
also fyi - there is a merge conflict - that'll still be there after master is fixed - u mind fixing that at ur convenience (if not, i'll do it when i fix the master drift, all good)

and btw - thanks for ur contribution - much appreciated :)

**Comment by arjunkathuria** [comment]
@eriktrom do we need this truthy or falsey ?

```
  1)  if (checker.forEmber().satisfies('>= 2.3')) {
=======
  2)  if (checker.forEmber().satisfies('>= 2.3') === false) {
```

former was in originally, PR 74 changed it to later, im guessing we want the later one, confirm please.

**Comment by eriktrom** [comment]
@arjunkathuria - so good question - mostly b/c the previous code seems wrong (unless im missing something)

We need to include the get-owner-polyfill in this addon when the version of ember in the consuming app does not provide built in support for `Ember.getOwner`.

Basically, this tells me the previous lines - `if (checker.for('ember', 'bower').satisfies('>= 2.3')) {` we're actually incorrect - I don't know when `getOwner` shipped, but assuming it shipped in Ember 2.4 (stable) then we should change that to:

```js
if (checker.forEmber().satisfies('< 2.4')) {
  addonPackages.push({name: 'ember-getowner-polyfill', target: '^1.0.0'});
}
```

which would add the polyfill for consumers of this addon that are using Ember < 2.4. 

(I'm not sure when `getOwner` dropped though - u may want to double check that)

tl;dr - your `=== false` version was correct - but lets use `< 2.4` for better clarity

thanks again @arjunkathuria

**Comment by arjunkathuria** [comment]
@eriktrom `getOwner` was introduced in Ember 2.3 :- [source](http://emberjs.com/deprecations/v2.x/#toc_injected-container-access)

Thus, anything below 2.3 `('< 2.3')` would need the polyfill.

Changes made to reflect the same.

**Comment by rwwagner90** [comment]
Isn't the point of a polyfill to make all versions work? You should always include ember-getowner-polyfill.

**Comment by eriktrom** [comment]
@rwwagner90 

> Isn't the point of a polyfill to make all versions work? You should always include ember-getowner-polyfill.

this was my initial thought - but b/c we're just updating code that was already there (which checks ember-source as well as bower ember) my thought was 'there must be a reason'

if there is not a reason someone can explain, should we remove this code instead?

in ember 3.0.0 will the polyfill remain supported, or is it only for ember < 3.x is one question is the only real reason i can think of

anyone is free to comment further, i don't have the context here

**Comment by eriktrom** [comment]
@rwwagner90 - reason - bits over the wire reduced, few as they may be
(best guess for original code, based on runspired's perf first mentality)

@arjunkathuria 👍

**Comment by rwwagner90** [comment]
@eriktrom I would just ship the polyfill. I think it is quite small, but have no actually looked into it.
